accepted/future-releases/primary-constructors/feature-specification.md

```diff
@@ -745,6 +747,27 @@ main() {
 }
 ```
 
+A compile-time error occurs if an assignment to a primary parameter occurs
+in the initializing expression of a non-late instance variable, or in the
+initializer list of the body part of a primary constructor.
+
+*This includes expressions like `p++` where the assignment is implicit.
+The rule only applies for non-late variables because the primary parameters
```

Comment:
Consider:

-> The rule does not apply to late variables or static variables. The primary constructor parameters are not in scope for initializer expressions of those variables at all.

(More direct and mentions static variables, redundantly since the above says "instance variables", but the above also says "non-late", so this commentary might as well be just as complete.)